Fix zypp dependency on Python 3.5

In [CI we get](https://dev.azure.com/lowrisc/ibex/_build/results?buildId=7828&view=logs&j=e92b124b-3ef9-5461-84e3-a5c4020c184a&t=c6688988-a2fe-54a8-1baf-ba20316d2af5)

```
Traceback (most recent call last):
  File "/usr/local/bin/fusesoc", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/usr/local/lib/python3.5/dist-packages/pkg_resources/__init__.py", line 3253, in <module>
    @_call_aside
  File "/usr/local/lib/python3.5/dist-packages/pkg_resources/__init__.py", line 3237, in _call_aside
    f(*args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/pkg_resources/__init__.py", line 3266, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/usr/local/lib/python3.5/dist-packages/pkg_resources/__init__.py", line 584, in _build_master
    ws.require(__requires__)
  File "/usr/local/lib/python3.5/dist-packages/pkg_resources/__init__.py", line 901, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/local/lib/python3.5/dist-packages/pkg_resources/__init__.py", line 787, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'zipp>=0.5' distribution was not found and is required by importlib-metadata
##[error]Verilog lint failed. Run 'fusesoc --cores-root . run --target=lint lowrisc:ibex:ibex_core_tracing' to check and fix all errors.
```

The root cause is that the zipp library did a release that doesn't support Python 3.5 any more. It's not quite clear to me yet why the dependency resolution logic in pip chooses this version then, and doesn't stick with an older version. 

I've seen the same problem (with a slightly different error message) in OpTiMSoC as well for ~2 weeks and hoped it would be resolved by just waiting, turns out that's not the case.

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -31,6 +31,12 @@ jobs:
   # Installing six is a workaround for pip dependency resolution: six is already
   # installed as system package with a version below the required one.
   # Explicitly installing six through pip gets us a supported version.
+  #
+  # Updating pip and setuptools is required to have these tools properly parse
+  # Python-version metadata, which some packages uses to specify that an older
+  # version of a package must be used for a certain Python version. If that
+  # information is not read, pip installs the latest version, which then fails
+  # to run.
   - bash: |
       sudo apt-get install -y \
           python3 \
@@ -47,7 +53,8 @@ jobs:
           curl \
           libelf-dev \
           clang-format \
-        && sudo pip3 install -U six fusesoc
+        && sudo pip3 install -U setuptools pip six \
+        && sudo pip3 install -U fusesoc
     displayName: Install dependencies
 
   - bash: |

